.github/workflows/roll_torch_mlir.yml

```diff
@@ -0,0 +1,60 @@
+name: Roll PyTorch
+
+on:
+  schedule:
+    - cron: '0 13 * * *'
+  workflow_dispatch:
+
+jobs:
+  build_linux:
+    name: roll torch-mlir
+    needs: setup
+    runs-on:
+      - self-hosted
+      - runner-group=${{ needs.setup.outputs.runner-group }}
+      - environment=${{ needs.setup.outputs.runner-env }}
+      - cpu
+      - os-family=Linux
+
+    # Don't run this in everyone's forks.
+    if: github.repository == 'open-xla/iree'
+
+    steps:
+
+      - name: Prepare workspace
+        run: |
+          # Clear the workspace directory so that we don't run into errors about
+          # existing lock files.
+          sudo rm -rf $GITHUB_WORKSPACE/*
+
+      - name: Get iree
+        uses: actions/checkout@v3
+        with:
+          submodules: 'true'
+          token: ${{ secrets.WRITE_ACCESS_TOKEN }}
```

Comment:
I think we don't need to set a token as by default it will use a [github token from the workflow](https://docs.github.com/en/actions/security-guides/automatic-token-authentication). Here is an example we checkout code in another workflow:

https://github.com/openxla/iree/blob/bc2a32d47bd2c32571b9f830da4a71bb1211d63a/.github/workflows/ci.yml#L144-L145

Also it's recommended to pin the action version on a specific SHA (https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions)